openage/convert/service/export/png/binpack.pyx

```diff
@@ -212,41 +218,41 @@ cdef class BinaryTreePacker(Packer):
         self.mapping = {}
         self.root = NULL
 
-        for block in sorted(blocks, key=maxside_heuristic, reverse=True):
-            self.fit(block)
+        for i, block in enumerate(sorted(blocks, key=maxside_heuristic, reverse=True)):
+            self.fit(i, block)
```

Comment:
The problem here is that index `i` is the index in the list of the already sorted blocks, not the index of the original list of blocks. You then insert `i` into the mapping in `fit(..)`. When you later call `pos(index)`, it still uses the original block index for the lookup even though `mapping` maps the sorted indices.

This probably also cause your image size being wrong since that depends on `pos(..)`.

Reply:
You should try to preserve the original index during sorting, e.g. by extending the tuple of a block to `(width, height, index)`